crates/swc_ecma_transforms_typescript/src/strip.rs

```diff
@@ -330,14 +330,16 @@ where
             }
 
             Decl::TsInterface(TsInterfaceDecl { ref id, .. })
-            | Decl::TsModule(TsModuleDecl {
-                id: TsModuleName::Ident(ref id),
-                ..
-            })
             | Decl::TsTypeAlias(TsTypeAliasDecl { ref id, .. }) => {
                 self.store(id.sym.clone(), id.span.ctxt, false)
             }
 
+            Decl::TsModule(TsModuleDecl {
+                id: TsModuleName::Ident(ref id),
+                declare,
+                ..
+            }) => self.store(id.sym.clone(), id.span.ctxt, !declare),
```

Comment:
I think this code is correct if declare is true, but we have to mark it as `type + var` if declare is false.